packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/project/controllers/MutationUploadProjectDataFetcher.java

```diff
@@ -42,35 +46,50 @@ public class MutationUploadProjectDataFetcher implements IDataFetcherWithFieldCo
 
     private static final String FILE = "file";
 
+    private final List<ICapabilityVoter> capabilityVoters;
+
     private final IProjectImportService projectImportService;
 
     private final IMessageService messageService;
 
-    public MutationUploadProjectDataFetcher(IProjectImportService projectImportService, IMessageService messageService) {
+    public MutationUploadProjectDataFetcher(List<ICapabilityVoter> capabilityVoters, IProjectImportService projectImportService, IMessageService messageService) {
+        this.capabilityVoters = Objects.requireNonNull(capabilityVoters);
         this.projectImportService = Objects.requireNonNull(projectImportService);
         this.messageService = Objects.requireNonNull(messageService);
     }
 
     @Override
     public IPayload get(DataFetchingEnvironment environment) throws Exception {
+        IPayload payload = new ErrorPayload(UUID.randomUUID(), this.messageService.unexpectedError());
         Map<Object, Object> input = environment.getArgument(INPUT_ARGUMENT);
 
-        var optionalId = Optional.ofNullable(input.get(ID))
+        var optionalId = Optional.ofNullable(input)
+                .map(map -> map.get(ID))
                 .map(Object::toString)
                 .flatMap(new UUIDParser()::parse);
 
-        var optionalFile = Optional.ofNullable(input.get(FILE))
+        var optionalFile = Optional.ofNullable(input)
+                .map(map -> map.get(FILE))
                 .filter(UploadFile.class::isInstance)
                 .map(UploadFile.class::cast);
 
-        if (optionalId.isPresent() && optionalFile.isPresent()) {
-            var id = optionalId.get();
-            var uploadFile = optionalFile.get();
-            return this.projectImportService.importProject(id, uploadFile);
+        if (optionalId.isPresent()) {
+            var inputId = optionalId.get();
+            payload = new ErrorPayload(inputId, this.messageService.unexpectedError());
+
+            if (optionalFile.isPresent()) {
```

Comment:
There's no need to perform both check in separate parts

Reply:
- None of those field can be null, that's not a reachable state in the GraphQL API
- There shouldn't be any difference between optionalId and optionalFile, consider them both at the same time